WooCommerce/Classes/GoogleAds/GoogleAdsEligibilityChecker.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import Experiments
+import Yosemite
+
+/// Interface for checking if a site is eligible for creating Google ads campaigns from the app.
+///
+protocol GoogleAdsEligibilityChecker {
+    @MainActor
+    func isSiteEligible(siteID: Int64) async -> Bool
+}
+
+final class DefaultGoogleAdsEligibilityChecker: GoogleAdsEligibilityChecker {
+
+    private let stores: StoresManager
+    private let featureFlagService: FeatureFlagService
+
+    init(stores: StoresManager = ServiceLocator.stores,
+         featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService) {
+        self.stores = stores
+        self.featureFlagService = featureFlagService
+    }
+
+    func isSiteEligible(siteID: Int64) async -> Bool {
+        guard featureFlagService.isFeatureFlagEnabled(.googleAdsCampaignCreationOnWebView) else {
+            return false
+        }
+
+        let isPluginSatisfied: Bool = await {
+            if let savedPlugin = await fetchPluginFromStorage(siteID: siteID) {
+                return checkIfGoogleAdsIsSupported(plugin: savedPlugin)
```

Comment:
❓ Should we fetch the plugin from remote in case the plugin version has been updated recently and it isn't synched to storage yet? 
This is an edge case considering that we fetch plugin info upon app launch.

Reply:
@selanthiraiyan do you mean we should always fetch from remote rather than relying on storage?

Reply:
> do you mean we should always fetch from remote rather than relying on storage?

We can fetch from remote only when the local plugin version doesn't match the required `pluginMinimumVersion`. This covers the case in which the user has updated the plugin version recently but we haven't fetched the new version details. This is an edge case and please feel free to ignore it.

Reply:
I think it is safer to always fetch plugins from remote to avoid edge cases. Updated in 5f0f3ea14219cfd104ddada332464d25c27d5f1e. Thanks Sharma for the suggestion!